### PR TITLE
View request response builders

### PR DESCRIPTION
**Pull request by jdillon**
Adds Request.Builder, Response.Builder and removes PayloadResponse (Request now has Payload)

### CODE REVIEW

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF367-2 :white_check_mark:

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/partial/PartialFetchHandler.java [hunk 1]
```diff
@@ -102,37 +98,37 @@ public Response handle(@Nonnull final Context context) throws Exception {
...
+                                   final Range<Long> requestedRange)
   {
-    response.setStatus(Status.success(HttpStatus.PARTIAL_CONTENT));
-    final Range<Long> rangeToSend = requestedRange;
+    Response.Builder builder = new Response.Builder()
+        .copy(response)
+        .status(Status.success(HttpStatus.PARTIAL_CONTENT));
 
+    // FIXME: Remove pointless local
+    final Range<Long> rangeToSend = requestedRange;
```
Comment (mcculls) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/partial/PartialFetchHandler.java]: Why not just remove this and use `requestedRange` directly?

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/Context.java [hunk 2]
```diff
@@ -97,6 +86,21 @@ public Response proceed() throws Exception {
   }
 
   //
+  // Attributes
+  //
+
+  /**
+   * Custom attributes to configure backing and logging.
+   */
+  public static class Attributes
```
Comment (mcculls) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/Context.java]: This change appears unrelated to the PR title "request response builders" - any particular reason to make this public? Also, since `AttributesMap`'s default constructor already creates a new hash map, why not use that?
```
public static class Attributes
    extends AttributesMap
{
   // custom attributes go here?
}
```

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/Request.java [hunk 3]
```diff
@@ -109,13 +101,156 @@ public boolean isMultipart() {
...
+  //
+
+  /**
+   * Custom attributes to configure backing and logging.
+   */
+  public static class Attributes
+      extends AttributesMap
+  {
+    public Attributes() {
+      super(Maps.<String, Object>newHashMap());
```
Comment (mcculls) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/Request.java]: As above, why use this this instead of `AttributesMap`'s default constructor which does the same thing?

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/Response.java [hunk 4]
```diff
@@ -12,25 +12,76 @@
...
+
+  //
+  // Attributes
+  //
 
   /**
    * Custom attributes to configure backing and logging.
    */
-  private static class Attributes
+  public static class Attributes
```
Comment (mcculls) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/Response.java]: I don't see a method that returns this type, so is this being made public for testing or some other purpose?

File: plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/describe/DescriptionHelper.java [hunk 5]
```diff
@@ -47,7 +46,6 @@ public void describeRequest(final Description desc, final Request request) {
 
     desc.addTable("Details", ImmutableMap.<String, Object>builder()
             .put("Action", request.getAction())
-            .put("URL", request.getRequestUrl())
```
Comment (mcculls) [inline plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/describe/DescriptionHelper.java]: I assume this is meant to be removed?

**Comment by mcculls** [comment]
+1  ...especially the use of fluent API ;)

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/Request.java (hunk 3 above)
Comment (jdillon) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/Request.java]: Mostly just for logging so logs show org.sonatype.nexus.repository.view.Request.Attributes instead of org.sonatype.nexus.common.collect.AttributeMap.

It could however use the default ctor, think that came later than when this was originally written.

I made these public, now that we have a builder pattern, in case someone wanted to prime the attributes with an object, instead of using the default helpers.  Otherwise it could be private again.  /me shrugs.

File: plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/describe/DescriptionHelper.java (hunk 5 above)
Comment (jdillon) [inline plugins/basic/nexus-repository-httpbridge/src/main/java/org/sonatype/nexus/repository/httpbridge/internal/describe/DescriptionHelper.java]: yes, request no longer carries the URL

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/partial/PartialFetchHandler.java (hunk 1 above)
Comment (jdillon) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/partial/PartialFetchHandler.java]: Yes I could have :-P

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/Response.java [hunk 6]
```diff
@@ -38,40 +89,85 @@ public Attributes() {
...
+    public Builder attributes(final AttributesMap attributes) {
+      this.attributes = attributes;
+      return this;
+    }
 
-  public void setStatus(final Status status) {
-    this.status = checkNotNull(status);
-  }
+    @Nonnull
+    public AttributesMap attributes() {
```
Comment (mrprescott) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/Response.java]: Should this be .getAttributes(), being that it's not part of the fluent stuff?

**Comment by mrprescott** [comment]
+1

**Comment by jdillon** [comment]
manually merged
